yaptide/batch/batch_methods.py

```diff
@@ -1,26 +1,85 @@
+from fabric import Connection, Result
+from paramiko import Ed25519Key
+
 import json as json_lib
 import os
+import sys
+import tempfile
 
-import time
 from datetime import datetime
 
 from pathlib import Path
 
-from yaptide.persistence.models import SimulationModel
+from yaptide.persistence.models import SimulationModel, ClusterModel
+
+from yaptide.batch.string_templates import SHIELDHIT_BASH
+
+# dirty hack needed to properly handle relative imports in the converter submodule
+sys.path.append("yaptide/converter")
+from ..converter.converter.api import get_parser_from_str, run_parser  # skipcq: FLK-E402
+
 
 ROOT_DIR = os.path.dirname(os.path.realpath(__file__))
 
 
-def submit_job(json_data: dict) -> tuple[dict, int]:  # skipcq: PYL-W0613
+def write_input_files(json_data: dict, output_dir: Path):
+    """
+    Function used to write input files to output directory.
+    Returns dictionary with filenames as keys and their content as values
+    """
+    if "beam.dat" not in json_data["sim_data"]:
+        conv_parser = get_parser_from_str(json_data["sim_type"])
+        return run_parser(parser=conv_parser, input_data=json_data["sim_data"], output_dir=output_dir)
+
+    for key, file in json_data["sim_data"].items():
+        with open(Path(output_dir, key), "w") as writer:
+            writer.write(file)
+    return json_data["sim_data"]
+
+
+def submit_job(json_data: dict, cluster: ClusterModel) -> tuple[dict, int]:  # skipcq: PYL-W0613
     """Dummy version of submit_job"""
+    with tempfile.TemporaryDirectory() as tmp_dir_path:
+        input_files = write_input_files(json_data, Path(tmp_dir_path))
+        script = SHIELDHIT_BASH.format(
+            beam=input_files["beam.dat"],
+            detect=input_files["detect.dat"],
+            geo=input_files["geo.dat"],
+            mat=input_files["mat.dat"]
+        )
+        ssh_key_path = Path(tmp_dir_path, "id_ed25519")
```

Comment:
do we need to write key to the file ? can't we pass it as a string ?

Reply:
That was my initial target and I was trying to find the solution but nothing else worked. I will work on it at the end of the PR because more important things are going to be added.